### PR TITLE
#3280 Contracts: Make sure that if extending process crashes while extending, doesn't rollback all 

### PR DESCRIPTION
**Pull request by cristinamghita**
#3280 Contracts: Make sure that if extending process crashes while extending, doesn't rollback all

### CODE REVIEW

File: de.metas.contracts/src/main/java/de/metas/contracts/impl/FlatrateBL.java [hunk 1]
```diff
@@ -1006,46 +1007,25 @@ else if (diffPercent.signum() < 0 && diffPercent.abs().compareTo(conditions.getM
...
 	@Override
-	public void extendContract(
-			final I_C_Flatrate_Term currentTerm,
-			final boolean forceExtend,
-			final Boolean forceComplete,
-			final Timestamp nextTermStartDate,
-			final I_C_OrderLine ol)
+	public void extendContract(final @NonNull ContractExtendingRequest context)
 	{
+		Services.get(ITrxManager.class).run(ITrx.TRXNAME_ThreadInherited, (TrxRunnable)localTrxName -> {
```
Comment (metas-ts) [inline de.metas.contracts/src/main/java/de/metas/contracts/impl/FlatrateBL.java]: i think we don't need the cast to `TrxRunnable`

File: de.metas.contracts/src/main/java/de/metas/contracts/IFlatrateBL.java [hunk 2]
```diff
@@ -89,17 +93,30 @@
 	void updateEntry(I_C_Flatrate_DataEntry dataEntry);
 
 	/**
-	 * Create a new flatrate term using the given term as template. The new term's C_Year will be the year after the given term's C_Year.
-	 *
-	 * <b>IMPORTANT:</b> the method might set the given term's C_FlatrateTerm_Next_ID, but won't save it!
+	 * term to extend
+	 * forceExtend - will create a new term, even if the given <code>term</code> has <code>IsAutoRenew='N'</code>
+	 * forceComplete - will complete a new term (if one has been created), even if it has <code>IsAutoComplete='N'</code>
```
Comment (metas-ts) [inline de.metas.contracts/src/main/java/de/metas/contracts/IFlatrateBL.java]: here is an outdated (a.k.a. lying) comment

Comment (cristinamghita) [reply de.metas.contracts/src/main/java/de/metas/contracts/IFlatrateBL.java]: yes, right!
Thx!

File: de.metas.contracts/src/main/java/de/metas/contracts/impl/FlatrateBL.java (hunk 1 above)
Comment (cristinamghita) [reply de.metas.contracts/src/main/java/de/metas/contracts/impl/FlatrateBL.java]: Yes, is added by default by eclipse, due to a setting

**Pull request merged**